hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleQuantiles.java

```diff
@@ -243,7 +245,12 @@ synchronized public Map<Quantile, Long> snapshot() {
     
     Map<Quantile, Long> values = new TreeMap<Quantile, Long>();
     for (int i = 0; i < quantiles.length; i++) {
-      values.put(quantiles[i], query(quantiles[i].quantile));
+      /* eg : effectiveQuantile for 0.99 with inverseQuantiles will be 0.01. 
+      For inverse quantiles higher numeric value is better and hence we want 
+      to query from the opposite end of the sorted sample
+       */
+      double effectiveQuantile = inverseQuantiles ? 1 - quantiles[i].quantile : quantiles[i].quantile;
+      values.put(quantiles[i], query(effectiveQuantile));
```

Comment:
Wouldn't reversing list order traversal give you the same thing wihtout altering the math?

Reply:
Inversing the traversal does not give us the same thing. The iterator is a variable for calculating the allowableError, which becomes high if the traversal is inverted. It will require more math in the allowableError() method based on [CKMS paper.](http://dimacs.rutgers.edu/~graham/pubs/papers/bquant-icde.pdf)

Looking for the inverse quantile (1 - quantile) however works as expected as shown in the UT runs from the PR description.